src/replication.c

```diff
@@ -476,19 +479,44 @@ static void appendLeaderCb(struct raft_io_append *req, int status)
 
     /* In case of a failed disk write, if we were the leader creating these
      * entries in the first place, truncate our log too (since we have appended
-     * these entries to it) and fire the request callback.
+     * these entries to it) and fire the request callbacks.
      *
      * Afterward, convert immediately to follower state, giving the cluster a
      * chance to elect another leader that doesn't have a full disk (or whatever
      * caused our write error). */
     if (status != 0) {
-        struct raft_apply *apply;
         ErrMsgTransfer(r->io->errmsg, r->errmsg, "io");
-        apply =
-            (struct raft_apply *)getRequest(r, request->index, RAFT_COMMAND);
-        if (apply != NULL) {
-            if (apply->cb != NULL) {
-                apply->cb(apply, status, NULL);
+        for (unsigned i = 0; i < request->n; i++) {
+            const struct request *req = getRequest(r, request->index + i, -1);
+            if (!req) {
```

Comment:
not sure if we should assert `req` here, but the original code also ignored `NULL` requests.